lib/daisy_ui_components/loading.ex

```diff
@@ -0,0 +1,54 @@
+defmodule DaisyUIComponents.Loading do
+  use DaisyUIComponents.Component
+
+  attr :class, :string, default: nil
+  attr :id, :string, required: true
+  attr :loading, :boolean, default: false
+  attr :shape, :string, required: true, values: ~w(spinner dots ring ball bars infinity)
+  attr :color, :string, values: colors() ++ ["neutral"]
+  attr :size, :string, values: sizes()
+
+  def loading(assigns) do
+    assigns =
+      assign(assigns, :class, [
+        "loading",
+        loading_shape(assigns[:shape]),
+        loading_color(assigns[:color]),
+        loading_size(assigns[:size]),
+        assigns.class
+      ])
+
+    ~H"""
+    <%= if @loading == true do %>
```

Comment:
I think this `if` is not necessary because users can still do this directly on the component:

```elixir
<DaisyUIComponents.Loading.loading :if={@loading} />
```

Reply:
[addressing some comments](https://github.com/phcurado/daisy_ui_components/pull/31/commits/98ea99728317a9c5cbaba6ca4f060ea05235c413)